src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs

```diff
@@ -18,47 +17,17 @@ public MainPipelineExecutor(IServiceProvider rootBuilder, IPipelineCache pipelin
             this.receivePipeline = receivePipeline;
         }
 
-        public async Task Invoke(MessageContext messageContext, CancellationToken cancellationToken = default)
+        public Task Invoke(MessageContext messageContext, CancellationToken cancellationToken = default)
         {
-            var pipelineStartedAt = DateTimeOffset.UtcNow;
-
-            using (var childScope = rootBuilder.CreateScope())
-            {
-                var message = new IncomingMessage(messageContext.NativeMessageId, messageContext.Headers, messageContext.Body);
-
-                var rootContext = new RootContext(childScope.ServiceProvider, messageOperations, pipelineCache, cancellationToken);
-                rootContext.Extensions.Merge(messageContext.Extensions);
-
-                var transportReceiveContext = new TransportReceiveContext(message, messageContext.TransportTransaction, rootContext);
-
-                try
-                {
-                    await receivePipeline.Invoke(transportReceiveContext).ConfigureAwait(false);
-                }
-#pragma warning disable PS0019 // Do not catch Exception without considering OperationCanceledException - enriching and rethrowing
-                catch (Exception ex)
-#pragma warning restore PS0019 // Do not catch Exception without considering OperationCanceledException
-                {
-                    ex.Data["Message ID"] = message.MessageId;
-
-                    if (message.NativeMessageId != message.MessageId)
-                    {
-                        ex.Data["Transport message ID"] = message.NativeMessageId;
-                    }
-
-                    ex.Data["Pipeline canceled"] = transportReceiveContext.CancellationToken.IsCancellationRequested;
-
-                    throw;
-                }
-
-                await receivePipelineNotification.Raise(new ReceivePipelineCompleted(message, pipelineStartedAt, DateTimeOffset.UtcNow), cancellationToken).ConfigureAwait(false);
-            }
+            return Task.CompletedTask;
```

Comment:
If this throws an exception the stack trace gets all messed up. Suggest using `async`/`await` here:

```csharp
public async Task Invoke(MessageContext messageContext, CancellationToken cancellationToken = default)
     => await Task.CompletedTask;
```

Reply:
Great suggestion!